driver/softmaxv3_driver.hpp

```diff
@@ -0,0 +1,552 @@
+        return miopenStatusInternalError;
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int SoftmaxV3Driver<Tgpu, Tref>::RunForwardCPU()
+{
+    mloSoftmaxV3ForwardContiguousRunHost(inputTensor, in.data(), outhost.data(), dim, algorithm);
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int SoftmaxV3Driver<Tgpu, Tref>::RunBackwardGPU()
+{
+    float kernel_total_time = 0;
+    float kernel_first_time = 0;
+    Timer t;
+    START_TIME;
+    for(int i = 0; i < inflags.GetValueInt("iter"); i++)
+    {
+        miopenStatus_t status = miopenSoftmaxBackward_V3(GetHandle(),
+                                                         outputTensor,
+                                                         out_dev->GetMem(),
+                                                         outGradTensor,
+                                                         outGrad_dev->GetMem(),
+                                                         inGradTensor,
+                                                         inGrad_dev->GetMem(),
+                                                         dim,
+                                                         algorithm);
+
+        MIOPEN_THROW_IF(status != miopenStatusSuccess, "Error in miopenSoftmaxBackward_V3");
+
+        float time = 0.0;
+        miopenGetKernelTime(GetHandle(), &time);
+        kernel_total_time += time;
+        if(i == 0)
+            kernel_first_time = time;
+    }
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        STOP_TIME
+        int iter = inflags.GetValueInt("iter");
+        if(WALL_CLOCK)
+            std::cout << "Wall-clock Time Backward Softmax Elapsed: " << t.gettime_ms() / iter
+                      << " ms\n";
+        float kernel_average_time =
+            iter > 1 ? (kernel_total_time - kernel_first_time) / (iter - 1) : kernel_first_time;
+        std::cout << "GPU Kernel Time Backward Softmax Elapsed: " << kernel_average_time << " ms\n";
+    }
+
+    if(inGrad_dev->FromGPU(GetStream(), inGrad.data()) != 0)
+        std::cerr << "Error copying (out_dev) from GPU, size: " << inGrad_dev->GetSize()
+                  << std::endl;
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+Tref SoftmaxV3Driver<Tgpu, Tref>::GetTolerance()
+{
+    Tref tolerance = std::numeric_limits<Tgpu>::epsilon() * 10;
+    return tolerance;
+}
+
+template <typename Tgpu, typename Tref>
+int SoftmaxV3Driver<Tgpu, Tref>::VerifyForward()
+{
+    RunForwardCPU();
+    const Tref tolerance = GetTolerance();
+    auto error           = miopen::rms_range(outhost, out);
+
+    if(!std::isfinite(error) || error > tolerance)
+    {
+        std::cout << "Forward Softmax FAILED: " << error << " > " << tolerance << std::endl;
+        return EC_VerifyFwd;
+    }
+    else
+    {
+        std::cout << "Forward Softmax Verifies OK on CPU reference (" << error << " < " << tolerance
```

Comment:
Based on your condition (`error > tolerance`), this actually should be `<=`, but it's ok
```suggestion
        std::cout << "Forward Softmax Verifies OK on CPU reference (" << error << " <= " << tolerance
```

Reply:
updated.

Reply:
you didnt updated it, but I think using `<` is not big deal, so let's just keep it as it is

Reply:
you updated it in backward mode, update it in forward mode too

Reply:
re-updated.